vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -34,8 +34,12 @@ def authentication_password_encrypted(type = nil)
     authentication_component(type, :password_encrypted)
   end
 
+  def required_credential_fields(type)
+    [:userid]
+  end
+
   def has_credentials?(type = nil)
-    !!authentication_component(type, :userid)
+    required_credential_fields(type).all?{ |field|  !!authentication_component(type, field) }
```

Comment:
Fixed :) Although I must say I prefer being explicit with `!!` unless the method has a trailing `?`